.github/workflows/staticcheck.yml

```diff
@@ -0,0 +1,19 @@
+name: "staticcheck"
+on: ["push", "pull_request"]
+
+jobs:
+  ci:
+    name: "Run Staticcheck"
+    runs-on: ubuntu-latest
```

Comment:
I think we want `continue-on-error: true` on this level too: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idcontinue-on-error